tests/manage/z_cluster/nodes/test_worker_nodes_network_failures.py

```diff
@@ -0,0 +1,223 @@
+
+        ceph_cluster = CephCluster()
+
+        # Label nodes to be able to run app pods
+        helpers.label_worker_node(
+            node_list=app_pod_nodes, label_key="nodetype", label_value="app-pod"
+        )
+
+        # Create deploymentconfig based pods for both interface
+        pod_objs = list()
+        interface_list = [constants.CEPHBLOCKPOOL, constants.CEPHFILESYSTEM]
+        logger.info(f"Starting app pods")
+        for num in range(self.num_of_app_pods):
+            for interface in interface_list:
+                pod_objs.append(
+                    dc_pod_factory(
+                        interface=interface, size=self.pvc_size,
+                        node_selector={'nodetype': 'app-pod'}
+                    )
+                )
+
+        return ceph_cluster, pod_objs, worker_nodes
+
+    @pytest.mark.parametrize(
+        argnames=["scenario"],
+        argvalues=[
+            pytest.param(
+                *['colocated'], marks=pytest.mark.polarion_id("OCS-1432")
+            ),
+            pytest.param(
+                *['dedicated'], marks=pytest.mark.polarion_id("OCS-1433")
+            )
+        ]
+    )
+    def test_all_worker_nodes_short_network_failure(self, nodes, setup):
+        """
+        OCS-1432/OCS-1433:
+        - Start DeploymentConfig based app pods
+        - Make all the worker nodes unresponsive by doing abrupt network failure
+        - Reboot the unresponsive node after short duration of ~300 seconds
+        - When unresponsive node recovers, app pods and ceph cluster should recover
+        - Verify data integrity and run IOs from app pods
+        """
+        ceph_cluster, pod_objs, worker_nodes = setup
+
+        # Run IO on pods
+        logger.info(f"Starting IO on {len(pod_objs)} app pods")
+        with ThreadPoolExecutor(max_workers=4) as executor:
+            for pod_obj in pod_objs:
+                logger.info(f"Starting IO on pod {pod_obj.name}")
+                executor.submit(
+                    pod_obj.run_io, storage_type='fs', size='1G',
+                    fio_filename='io_file1'
+                )
+        logger.info(f"IO started on all {len(pod_objs)} app pods")
+
+        # Verify IO results
+        md5sum_data = list()
+        for pod_obj in pod_objs:
+            pod.get_fio_rw_iops(pod_obj)
+            md5sum_data.append(
+                pod.cal_md5sum(pod_obj, 'io_file1')
+            )
+
+        logger.info(
+            f"Starting more IOs in background on {len(pod_objs)} app pods"
+        )
+        for pod_obj in pod_objs:
+            logger.info(f"Starting IO on pod {pod_obj.name}")
+            pod_obj.run_io(
+                storage_type='fs', size='1G', runtime=300,
+                fio_filename='bg_io_file'
+            )
+        logger.info(
+            f"IO started in background on all {len(pod_objs)} app pods"
+        )
+        sleep(20)  # Wait for io to run for 20 seconds
```

Comment:
can we poll the space being reclaimed and wait for a certain percentage to be filled instead of sleep?

Reply:
Removed sleep